service/src/main/java/uk/gov/hmcts/reform/fpl/service/ApplicationDocumentsService.java

```diff
@@ -0,0 +1,96 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.ApplicationDocument;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.DocumentUploadHelper;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static io.jsonwebtoken.lang.Collections.isEmpty;
+
+@Service
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class ApplicationDocumentsService {
+
+    private final Time time;
+    private final DocumentUploadHelper documentUploadHelper;
+
+    public Map<String, Object> updateCaseDocuments(List<Element<ApplicationDocument>> currentDocuments,
+                                                   List<Element<ApplicationDocument>> previousDocuments) {
+        List<Element<ApplicationDocument>> updatedDocuments = setUpdatedByAndDateAndTimeOnDocuments(
+             currentDocuments, previousDocuments);
+
+        Map<String, Object> updatedCaseData = new HashMap<>();
+
+        updatedCaseData.put("documents", updatedDocuments);
+
+        return updatedCaseData;
+    }
+
+    private List<Element<ApplicationDocument>> setUpdatedByAndDateAndTimeOnDocuments(
+        List<Element<ApplicationDocument>> currentDocuments,
+        List<Element<ApplicationDocument>> previousDocuments) {
+
+        if (isEmpty(previousDocuments) && !isEmpty(currentDocuments)) {
+            currentDocuments.stream().forEach(this::setUpdatedByAndDateAndTimeOnDocumentToCurrent);
```

Comment:
could be `currentDocuments.forEach(this::setUpdatedByAndDateAndTimeOnDocumentToCurrent)`